web/pgadmin/static/js/helpers/ModalProvider.jsx

```diff
@@ -104,7 +104,9 @@ function confirm(title, text, onOkClick, onCancelClick, okLabel = gettext('Yes')
       onOkClick?.();
       closeModal();
     };
-    const extraButtons =  extras?.(closeModal);
+    
+    const extraButtons = Array.isArray(extras) ? extras : extras?.(closeModal);
```

Comment:
why?

Reply:
We are sending an extra button **save and reload**  where we need that `closeModal` so this `extras?.(closeModal);` was only handling that but  In the delete button config which I created, I don't need closeModal.
That's why I updated that to handle both cases.

Reply:
The confirm dialog is so messed up now. Need to clean up.